Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Events.cs

```diff
@@ -237,6 +237,67 @@ public void Events_CanSwitchToProcessingInFixedUpdatesOnly()
         Assert.That(mouse.leftButton.isPressed, Is.True);
     }
 
+#if UNITY_EDITOR
+    // Test that changing focus enables or disables input processing, depending on the value of the
+    // runInBackground setting. Can only be effectively tested in the editor, because we don't have
+    // a generic means of setting player focuse (in the editor we can focus or unfocus the game window).
+    public enum Events_ShouldRunUpdate_TracksFocus_Mode
+    {
+        RunInBackground, DontRunInBackground
+    };
+    [Test]
+    [Category("Events")]
+    [TestCase(Events_ShouldRunUpdate_TracksFocus_Mode.RunInBackground)]
+    [TestCase(Events_ShouldRunUpdate_TracksFocus_Mode.DontRunInBackground)]
+    public void Events_ShouldRunUpdate_TracksFocus(Events_ShouldRunUpdate_TracksFocus_Mode runInBackgroundMode)
+    {
+        bool runInBackground = runInBackgroundMode == Events_ShouldRunUpdate_TracksFocus_Mode.RunInBackground;
+        InputSystem.settings.runInBackground = runInBackground;
+        Assert.That(InputSystem.settings.runInBackground, Is.EqualTo(runInBackground));
+        Assert.That(runtime.shouldRunInBackground, Is.EqualTo(runInBackground));
+        InputSystem.s_Manager.updateMask = InputUpdateType.Default;
+
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Dynamic));
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Fixed));
+        Assert.That(!runtime.onShouldRunUpdate.Invoke(InputUpdateType.Editor));
+
+        UnityEditor.EditorApplication.ExecuteMenuItem("Window/General/Scene");
+
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Dynamic), Is.EqualTo(runInBackground));
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Fixed), Is.EqualTo(runInBackground));
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Editor), Is.EqualTo(!runInBackground));
+
+        UnityEditor.EditorApplication.ExecuteMenuItem("Window/General/Game");
+
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Dynamic));
+        Assert.That(runtime.onShouldRunUpdate.Invoke(InputUpdateType.Fixed));
+        Assert.That(!runtime.onShouldRunUpdate.Invoke(InputUpdateType.Editor));
+    }
```

Comment:
Relying on window focus in a play mode test seems brittle. I think it's better to just test this at the `IInputRuntime` level. I.e. have `InputTestRuntime` go through focus changes and make sure the input system reacts properly.

Reply:
I agree that this is brittle, yeah. So you mean expose focus as a property of IInputRuntime instead (so we can override it in tests)? Yeah, I guess that makes sense.